File: Design Patterns/src/singleton/Tester.java
package singleton;


public class Tester {
    public static void main(String[] args) {
        IdGenerator idGen = IdGenerator.getInstance(0);
        System.out.println("IDGen Next Id: " + idGen.getNext());
        System.out.println("IDGen Next Id: " + idGen.getNext());


        IdGenerator idGen2 = IdGenerator.getInstance(1);
        System.out.println("IDGen2 Next Id: " + idGen2.getNext());
        System.out.println("IDGen2 Next Id: " + idGen2.getNext());

        IdGenerator idGen3 = IdGenerator.getInstance(0);
        System.out.println("IDGen3 Next Id: " + idGen3.getNext());
        System.out.println("IDGen3 Next Id: " + idGen3.getNext());
    }
}

File: Design Patterns/basic-patterns-java-kotlin/singleton/Tester.kt
package singleton


fun main() {
    val idGen = IdGenerator.getInstance(0)
    println("IDGen Next Id: " + idGen.next)
    println("IDGen Next Id: " + idGen.next)
    val idGen2 = IdGenerator.getInstance(1)
    println("IDGen2 Next Id: " + idGen2.next)
    println("IDGen2 Next Id: " + idGen2.next)
    val idGen3 = IdGenerator.getInstance(0)
    println("IDGen3 Next Id: " + idGen3.next)
    println("IDGen3 Next Id: " + idGen3.next)
}